cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -145,18 +145,23 @@ public KafkaBrokerConfigurationBuilder withCruiseControl(String clusterName, Cru
     /**
      * Configures the Strimzi Metrics Reporter. It is set only if user enables Strimzi Metrics Reporter.
      *
-     * @param model     Strimzi Metrics Reporter configuration
+     * @param model Strimzi Metrics Reporter configuration
      *
      * @return Returns the builder instance
      */
-    public KafkaBrokerConfigurationBuilder withStrimziMetricsReporter(MetricsModel model)   {
+    public KafkaBrokerConfigurationBuilder withStrimziMetricsReporter(MetricsModel model) {
+        printSectionHeader("Strimzi Metrics Reporter configuration");
         if (model instanceof StrimziMetricsReporterModel reporterModel) {
-            printSectionHeader("Strimzi Metrics Reporter configuration");
-            writer.println("prometheus.metrics.reporter.listener.enable=true");
-            writer.println("prometheus.metrics.reporter.listener=http://:" + StrimziMetricsReporterModel.METRICS_PORT);
-            writer.println("prometheus.metrics.reporter.allowlist=" + reporterModel.getAllowList());
-            writer.println();
+            writer.println(StrimziMetricsReporterConfig.LISTENER_ENABLE + "=true");
+            writer.println(StrimziMetricsReporterConfig.LISTENER + "=http://:" + MetricsModel.METRICS_PORT);
+            writer.println(StrimziMetricsReporterConfig.ALLOW_LIST + "=" + reporterModel.getAllowList());
+        } else {
+            // we disable the listener just in case the user manually set SMR from spec.config
+            // because this would enable the listener on port 8080, causing a node crash loop
+            // due to address already in use by the Kafka Agent
+            writer.println(StrimziMetricsReporterConfig.LISTENER_ENABLE + "=false");
```

Comment:
I would remove this. That would be a user misconfiguration. We cannot prevent every single thing like that. If user configures the Strimzi Metrics Reporter themself, it is also not clear whether they do not intend exactly what  you are trying to prevent.

Reply:
ok